srcdeps-core/src/main/java/org/l2x6/srcdeps/core/impl/scm/JGitScm.java

```diff
@@ -42,35 +55,141 @@
 @Named
 @Singleton
 public class JGitScm implements Scm {
-    public static final String SCM_GIT_PREFIX = "git:";
     private static final Logger log = LoggerFactory.getLogger(JGitScm.class);
+    public static final String SCM_GIT_PREFIX = "git:";
+    public static final String SRCDEPS_WORKING_BRANCH = "srcdeps-working-branch";
+
+    /**
+     * Tells if the given filesystem directory contains a valid git repository.
+     *
+     * @param dir
+     *            a {@link Path} to check
+     * @return {@code true} if the given {@code dir} contains a valid git repository; {@code false} otherwise.
+     */
+    private static boolean containsGitRepo(Path dir) {
+        Path gitDir = dir.resolve(".git");
+        if (!Files.exists(gitDir)) {
+            return false;
+        } else {
+            try (FileRepository repo = new FileRepository(gitDir.toFile())) {
+                return repo.getObjectDatabase().exists();
+            } catch (IOException e) {
+                log.warn(String.format("Could not check if [%s] contains a git repository", dir), e);
+                /*
+                 * We could perhaps throw e out of this method rather than return false. Returning false sounds as a
```

Comment:
yep, I think we should throw only if unexpected errors happen (file access permissions etc).